### PR TITLE
[TASK] Add TYPO3 v12 Support

### PR DESCRIPTION
**Pull request by achimfritz**
* Add TYPO3 v12 Support
* Update testing-framework
* Remove prophecy dependency
* Use CSV Fixtures

* Fixes: #225

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -5,14 +5,15 @@
     "homepage": "https://b13.com",
     "license": ["GPL-2.0-or-later"],
     "require": {
-        "typo3/cms-backend": "^10.4.18 || ^11.5",
-        "doctrine/dbal": "~2.13.1"
```
Comment (bmack) [inline composer.json:9]: I think we should keep the explicit dependency to doctrine/dbal (with ~2.13.1 || ~3.4)

File: Classes/Tca/ContainerConfiguration.php [hunk 2]
```diff
@@ -89,6 +92,12 @@ public function __construct(
         $this->label = $label;
         $this->description = $description;
         $this->grid = $grid;
+        if ((GeneralUtility::makeInstance(Typo3Version::class))->getMajorVersion() === 12) {
```
Comment (bmack) [inline Classes/Tca/ContainerConfiguration.php:95]: "=== 12" vs. "> 11" :)

File: Build/phpunit/FunctionalTests.xml [hunk 3]
```diff
@@ -7,8 +7,8 @@
     convertErrorsToExceptions="true"
     convertWarningsToExceptions="true"
     forceCoversAnnotation="false"
-    stopOnError="false"
-    stopOnFailure="false"
+    stopOnError="true"
+    stopOnFailure="true"
```
Comment (lolli42) [inline Build/phpunit/FunctionalTests.xml:11]: why? i tend to keep both false to see the full madness at once ;)

File: Classes/Tca/ContainerConfiguration.php (hunk 2 above)
Comment (achimfritz) [reply Classes/Tca/ContainerConfiguration.php:95]: hm, yes, but the filename is named `Partials12` , maybe we shoult change this when dropping v10 Support to `majorVersion === 11 then: Partials11` (but maybe Cores Partial File will change in 13 again, and we have to adapt our Partial again?, Nothing can be said, for this I fix the Condition and the Partial Name to `12`

**Review by bmack — APPROVED** [review]

**Pull request merged**